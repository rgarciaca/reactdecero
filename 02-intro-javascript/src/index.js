const activo = true;

let mensaje = "";

if (activo) {
  mensaje = "Activo";
} else {
  mensaje = "Inactivo";
}

console.log(mensaje);

const mensaje2 = activo ? "Activo 2" : "Inactivo 2";
console.log(mensaje2);
