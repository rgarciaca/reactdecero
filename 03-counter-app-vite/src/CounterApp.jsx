import { useState } from 'react';
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {

  const [counter, setCounter ] = useState(value);

  const handleAdd = ( e, modif ) => {
    setCounter( counter + modif );
  };

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <>
      <h2>Counter App</h2>
      <h3>{counter}</h3>

      <button onClick={(e) => handleAdd(e, -1)}>-1</button>
      <button onClick={(e) => handleAdd(e, 1)}>+1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: -1,
};
