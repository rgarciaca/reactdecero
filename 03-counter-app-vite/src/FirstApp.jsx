import PropTypes from "prop-types";

const newMessage = "Roberto Garcia";
const objeto = {
  message: "Hola Mundo",
  titulo: "Esto es un tipo de saludo",
};

const getSaludo = () => {
  return `Hola ${ newMessage }, ¿que tal estas?`;
};

export const FirstApp = ({ title, subtitle }) => {
  if (!title) throw new Error("El title no ha sido configurado.");

  return (
    <>
      <h2>{title}</h2>
      <p>{subtitle}</p>
      <code>{JSON.stringify(objeto)}</code>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: "Sin título",
  subtitle: "Sin subtítulo",
};
