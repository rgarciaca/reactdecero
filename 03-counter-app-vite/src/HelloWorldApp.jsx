// export function App() {
//     return <h1>Hola Mundo</h1>;
// }

export const HelloWorldApp = () => {
  return <h1>HelloWorlApp</h1>;
};
