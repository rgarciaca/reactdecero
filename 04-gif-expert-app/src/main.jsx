import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
    <div>Ejemplo del curso de Fernando Herrera - React: De cero a experto ( Hooks y MERN )</div>
  </React.StrictMode>,
)
