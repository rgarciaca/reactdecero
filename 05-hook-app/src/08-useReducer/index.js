export * from './TodoAdd';
export * from './TodoList';
export * from './TodoItem';
export * from './todoReducer';
export * from './TodoApp';
