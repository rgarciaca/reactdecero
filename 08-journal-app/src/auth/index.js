export * from './pages';
export * from './layout/AuthLayout';
export * from './routes/AuthRoutes';