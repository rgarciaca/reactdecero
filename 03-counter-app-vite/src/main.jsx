import React from "react";
import ReactDom from "react-dom/client";

import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp";

import "./main.css";

ReactDom.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <HelloWorldApp />
    <FirstApp title="Hola, soy Goku" subtitle="Soy un subtitulo!!!" />
    <CounterApp value={ 5 } />
  </React.StrictMode>
);
