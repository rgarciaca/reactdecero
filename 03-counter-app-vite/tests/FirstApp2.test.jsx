import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const subtitle = 'Soy un subtitulo!!!';
    const title = 'Hola, soy Goku';

    test("debe de hacer match con snapshot", () => {

        const { container}= render( <FirstApp title={ title }/>);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, soy Goku"', () => {

        render( <FirstApp title={ title }/>);
        expect( screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {

        render( <FirstApp title={ title }/>);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo por props', () => {

        render( <FirstApp title={ title } subtitle= {subtitle}/>);
        expect( screen.getByText(subtitle)).toBeTruthy();
    });
});