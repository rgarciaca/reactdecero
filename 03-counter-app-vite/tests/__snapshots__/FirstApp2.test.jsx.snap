// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    Sin subt√≠tulo
  </p>
  <code>
    {"message":"Hola Mundo","titulo":"Esto es un tipo de saludo"}
  </code>
</div>
`;
